app/routes/_layout+/kits.$kitId.manage-assets.tsx

```diff
@@ -0,0 +1,510 @@
+import { useEffect, useMemo } from "react";
+import { AssetStatus, type Prisma } from "@prisma/client";
+import { json, redirect } from "@remix-run/node";
+import type { ActionFunctionArgs, LoaderFunctionArgs } from "@remix-run/node";
+import { Form, useLoaderData, useNavigation } from "@remix-run/react";
+import { useAtom, useAtomValue } from "jotai";
+import { z } from "zod";
+import { kitsSelectedAssetsAtom } from "~/atoms/selected-assets-atoms";
+import { AssetImage } from "~/components/assets/asset-image";
+import DynamicDropdown from "~/components/dynamic-dropdown/dynamic-dropdown";
+import { FakeCheckbox } from "~/components/forms/fake-checkbox";
+import { ChevronRight } from "~/components/icons/library";
+import Header from "~/components/layout/header";
+import { List } from "~/components/list";
+import { Filters } from "~/components/list/filters";
+import { Button } from "~/components/shared/button";
+import { Image } from "~/components/shared/image";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "~/components/shared/tooltip";
+import { Td } from "~/components/table";
+import { db } from "~/database/db.server";
+import {
+  createBulkKitChangeNotes,
+  createNote,
+  getPaginatedAndFilterableAssets,
+} from "~/modules/asset/service.server";
+import { getUserByID } from "~/modules/user/service.server";
+import { makeShelfError, ShelfError } from "~/utils/error";
+import { isFormProcessing } from "~/utils/form";
+import { data, error, getParams, parseData } from "~/utils/http.server";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.validator.server";
+import { requirePermission } from "~/utils/roles.server";
+import { tw } from "~/utils/tw";
+
+export async function loader({ context, request, params }: LoaderFunctionArgs) {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  const { kitId } = getParams(params, z.object({ kitId: z.string() }), {
+    additionalData: { userId },
+  });
+
+  try {
+    const { organizationId } = await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.kit,
+      action: PermissionAction.update,
+    });
+
+    const [kit, assets] = await Promise.all([
+      db.kit
+        .findFirstOrThrow({
+          where: { id: kitId },
+          select: { id: true, name: true, assets: { select: { id: true } } },
+        })
+        .catch((cause) => {
+          throw new ShelfError({
+            cause,
+            title: "Kit not found!",
+            message:
+              "The kit you are trying to access does not exists or you do not have permission to asset it.",
+            status: 404,
+            label: "Kit",
+          });
+        }),
+      getPaginatedAndFilterableAssets({
+        request,
+        organizationId,
+        excludeSearchFromView: true,
```

Comment:
At some point I changed that because users were asking for a more powerful search on the manage-assets modal. I removed it already for bookings, so I think we have to skip it here. What you also need to make sure in that case that the search field looks like in the booking.manage-assets and has the little tooltip.